source/util.hpp

```diff
@@ -48,7 +48,14 @@ std::string indent(std::string const &code, std::string const &indentation);
 /// Try to read exisitng file and if content does not match to code - write a new version. Also create nested dirs starting from prefix if nessesary.
 void update_source_file(std::string const &prefix, std::string const &file_name, std::string const &code);
 
+/// Joins the vector into a string using the given separator
+std::string join( std::vector<std::string>& splits, std::string const& sep );
```

Comment:
again, `splits` should be taken by `const &`. Also i recommend to rename it something more descriptive, like `parts`

Reply:
I realized that after I checked in.